sys/dts/rpi3.dts

```diff
@@ -1,16 +1,71 @@
+/*
+ * Based on the following FreeBSD files:
+ *  - `sys/contrib/device-tree/src/arm/bcm2837.dtsi`
+ *  - `sys/contrib/device-tree/src/arm/bcm283x.dtsi`
+ */
+
 /dts-v1/;
 
 / {
-	#address-cells = <0x01>;
-	#size-cells = <0x01>;
+	#address-cells = <1>;
+	#size-cells = <1>;
 
 	memory@0 {
 		reg = <0x00 0x3c000000>;
 		device_type = "memory";
 	};
 
+	timer {
+		compatible = "arm,armv7-timer";
+		interrupt-parent = <&local_intc>;
+		interrupts = <0 4>, <1 4>, <2 4>, <3 4>;
+		always-on;
+	};
+
 	soc {
-		#address-cells = <0x01>;
-		#size-cells = <0x01>;
+		compatible = "simple-bus";
+		#address-cells = <1>;
+		#size-cells = <1>;
+		interrupt-parent = <&intc>;
+		ranges = <0x7e000000 0x3f000000 0x1000000>,
+			 <0x40000000 0x40000000 0x00001000>;
+		dma-ranges = <0xc0000000 0x00000000 0x3f000000>;
+
+		local_intc: local_intc@40000000 {
+			compatible = "brcm,bcm2836-l1-intc";
+			reg = <0x40000000 0x100>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			interrupt-parent = <&local_intc>;
+		};
+
+		intc: intc@7e00b200 {
+			compatible = "brcm,bcm2836-armctrl-ic";
+			reg = <0x7e00b200 0x200>;
+			interrupt-controller;
+			#interrupt-cells = <2>;
+			interrupt-parent = <&local_intc>;
+			interrupts = <8 4>;
+		};
+
+		uart0: serial@7e201000 {
+			compatible = "arm,pl011", "arm,primecell";
+			reg = <0x7e201000 0x200>;
+			interrupts = <2 25>;
+		};
+
+		sdhci: sdhci@7e300000 {
+			compatible = "brcm,bcm2835-sdhci";
+			reg = <0x7e300000 0x100>;
+			interrupts = <2 30>;
+		};
```

Comment:
One little caveat. In mimiker, we threat the sd controller as a generic emmc controller that provides a bus interface. Then we call methods from that interface using a child device that implements SD protocol. Thus I believe this should be renamed to `emmc` and `sdhci` should be used for a child node of `emmc`.
```suggestion
		emmc: emmc@7e300000 {
			compatible = "brcm,bcm2835-emmc";
			reg = <0x7e300000 0x100>;
			interrupts = <2 30>;
		};
```

Reply:
Also, you forgot to update the probe procedure of the emmc driver.

Reply:
Your remark is valid. Please put it into a separate PR together with better `probe` function that references FDT.